oommfc/drivers/hysteresisdriver.py

```diff
@@ -58,14 +70,44 @@ class HysteresisDriver(Driver):
         "report_wall_time",
     ]
 
-    def _checkargs(self, **kwargs):
-        Hmin, Hmax, n = kwargs["Hmin"], kwargs["Hmax"], kwargs["n"]
-        for i in [Hmin, Hmax]:
+    def _checkargs(self, kwargs):
+        # check the default arguments for a symmetric hysteresis loop
+        if any(item in kwargs for item in ["Hmin", "Hmax", "n"]) and "Hsteps" in kwargs:
+            msg = "Cannot define both (Hmin, Hmax, n) and Hsteps."
+            raise ValueError(msg)
+
+        if all(item in kwargs for item in ["Hmin", "Hmax", "n"]):
+            # case of a symmetric hysteresis simulation
+            self._checkvalues(kwargs["Hmin"], kwargs["Hmax"], kwargs["n"])
+            kwargs["Hsteps"] = [
+                [kwargs["Hmin"], kwargs["Hmax"], kwargs["n"]],
+                [kwargs["Hmax"], kwargs["Hmin"], kwargs["n"]],
+            ]
+        elif "Hsteps" in kwargs:
+            # case of multiple hysteresis sweep steps
+            if not isinstance(kwargs["Hsteps"], (list, tuple)):
+                raise TypeError("Hsteps has to be iterable.")
+            if any(len(element) != 3 for element in kwargs["Hsteps"]):
```

Comment:
```suggestion
            if any(not isinstance(element, (list, tuple)) or len(element) != 3 for element in kwargs["Hsteps"]):
```
Otherwise, this would fail with TypeError if an element has no `len` (e.g. an int or float).

Reply:
Yes, good idea,